sources/updater/update_metadata/src/lib.rs

```diff
@@ -296,58 +299,86 @@ impl Manifest {
+    pub fn update_ready(&self, seed: u32, time: DateTime<Utc>) -> bool {
+        // If this host is part of some update wave
         if let Some(wave) = self.update_wave(seed) {
-            return wave.has_started();
-        }
-
-        // Or there are no waves
-        true
-    }
-
-    pub fn jitter(&self, seed: u32) -> Option<DateTime<Utc>> {
-        if let Some(wave) = self.update_wave(seed) {
-            if wave.has_passed() {
-                return None;
-            }
-            let bounds = match self.update_wave(seed) {
-                Some(Wave::Initial { end }) => Some((Utc::now(), end)),
-                Some(Wave::General { start, end }) => Some((start, end)),
-                Some(Wave::Last { start: _ }) | None => None,
+            let bound = match wave {
+                // Hosts should not wind up in the special "initial" wave with no start time, but if they do,
+                // we consider the update as being available immediately.
+                Wave::Initial { .. } => None,
+                Wave::General {
+                    start_time,
+                    end_time,
+                    start_seed,
+                    end_seed,
+                } => Some((start_time, Some(end_time), start_seed, end_seed)),
+                // Last wave has no end time nor end seed; Let end seed be `MAX_SEED` since all the
+                // remaining hosts are in this last wave
+                Wave::Last {
+                    start_time,
+                    start_seed,
+                } => Some((start_time, None, start_seed, MAX_SEED)),
             };
-            if let Some((start, end)) = bounds {
-                let mut rng = thread_rng();
-                if let Some(range) = end.timestamp().checked_sub(start.timestamp()) {
-                    return Some(start + Duration::seconds(rng.gen_range(1, range)));
+            if let Some((start_time, maybe_end_time, start_seed, end_seed)) = bound {
+                // If the wave has passed, the update is available (this includes passing the last wave start time)
+                if wave.has_passed(time) {
+                    return true;
+                }
+                if let Some(end_time) = maybe_end_time {
+                    // This host is not part of last wave
+                    // Determine the duration of this host's wave
+                    let wave_duration = end_time - start_time;
+                    let num_seeds_allocated_to_wave = (end_seed - start_seed) as i32;
+                    if num_seeds_allocated_to_wave == 0 {
+                        // Empty wave, no host should have been allocated to it
+                        return true;
+                    }
+                    let time_increment = wave_duration / num_seeds_allocated_to_wave;
+                    // Derive the target time position within the wave given the host's seed.
+                    let target_time = start_time + (time_increment * (seed as i32));
```

Comment:
```suggestion
                    let time_per_seed = wave_duration / num_seeds_allocated_to_wave;
                    // Derive the target time position within the wave given the host's seed.
                    let target_time = start_time + (time_per_seed * (seed as i32));
```
Maybe it's a slightly clearer name?